ci/tests/puppeteer/run.sh

```diff
@@ -235,13 +267,56 @@ if [[ "${REBUILD}" == "true" && "${RERUN}" != "true" ]]; then
   FLAGS=$(echo $BUILDFLAGS | sed 's/ //')
   printgreen "\nBuilding CAS found in $PWD for dependencies [${dependencies}] with flags [${FLAGS}]"
 
-  ./gradlew :webapp:cas-server-webapp-${project}:build \
-    -DskipNestedConfigMetadataGen=true -x check -x javadoc --build-cache --configure-on-demand --parallel \
-    ${BUILD_SCRIPT} ${DAEMON} -DcasModules="${dependencies}" -q ${BUILDFLAGS}
+  BUILD_CACHE=""
+  if [[ "$GRADLE_BUILDCACHE_PSW" != "" || "$CI" != "true" ]]; then
+    BUILD_CACHE="--build-cache"
+  fi
 
-  if [ $? -eq 1 ]; then
+  printcyan "Launching build in background to make observing slow builds easier"
+  targetArtifact=./webapp/cas-server-webapp-${project}/build/libs/cas-server-webapp-${project}-${casVersion}.${projectType}
+  if [[ -d ./webapp/cas-server-webapp-${project}/build/libs ]]; then
+    rm -rf ./webapp/cas-server-webapp-${project}/build/libs
+  fi
+  buildcmd=$(printf '%s' \
+      "./gradlew :webapp:cas-server-webapp-${project}:build \
+      -DskipNestedConfigMetadataGen=true -x check -x test -x javadoc ${BUILD_CACHE} --configure-on-demand --parallel \
+      ${BUILD_SCRIPT} ${DAEMON} -DcasModules="${dependencies}" --no-watch-fs --max-workers=${MAX_WORKERS} ${BUILDFLAGS}")
+  echo $buildcmd
+  $buildcmd > build.log 2>&1 &
+  pid=$!
+  sleep 20
```

Comment:
I am not a fan of sleep. It would be faster/better if you could poll every 2 seconds. faster machines tend to complete much more quickly, and if I am done in 5 seconds, I should not have to wait an extra 15 seconds. Doing this with loop and a status check is more productive

Reply:
In this case the sleep is happening while the build is running in the background, so there isn't much waiting going on but some small percentage of the time on windows CI build (significant when there are 250+ tests) the java process or the gradle build seemed to be dying randomly so this gives it another chance. Checking for java running in a loop might allow it to restart immediately but I don't think the sleep is slowing down the normal case. I could change it but the extra complexity probably isn't warranted.

Reply:
Makes sense. Thanks for the note.